zmon_aws_agent/elastigroup.py

```diff
@@ -0,0 +1,197 @@
+    paginator = cf.get_paginator('list_stacks')
+    status_filter = [
+        "CREATE_COMPLETE",
+        "ROLLBACK_FAILED",
+        "DELETE_FAILED",
+        "UPDATE_IN_PROGRESS",
+        "UPDATE_COMPLETE_CLEANUP_IN_PROGRESS",
+        "UPDATE_COMPLETE",
+        "UPDATE_ROLLBACK_IN_PROGRESS",
+        "UPDATE_ROLLBACK_FAILED",
+        "UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS",
+        "UPDATE_ROLLBACK_COMPLETE"
+    ]
+
+    try:
+        response_iterator = call_and_retry(
+            lambda: paginator.paginate(StackStatusFilter=status_filter))
+        for page in response_iterator:
+            summaries = page['StackSummaries']
+            for summary in summaries:
+                stacks.append(summary['StackName'])
+        current_span.log_kv("num_stacks", len(stacks))
+    except Exception:
+        current_span.set_tag('error', True)
+        current_span.log_kv({'exception': traceback.format_exc()})
+        logger.exception('Failed to retrieve stack names')
+
+    return stacks
+
+
+@trace(pass_span=True)
+def get_elastigroup(elastigroup_data, **kwargs):
+    """
+    Returns the first Elastigroup that matches the id provided in the argument object.
+    """
+    current_span = extract_span_from_kwargs(**kwargs)
+    current_span.set_tag("cloud_account_id", elastigroup_data.account_id)
+    current_span.set_tag("elastigroup_id", elastigroup_data.group_id)
+    current_span.set_tag("elastigroup_name", elastigroup_data.group_name)
+    current_span.set_tag("span.kind", "client")
+
+    client = SpotinstClient(auth_token=elastigroup_data.access_token, account_id=elastigroup_data.account_id,
+                            print_output=False)
+
+    try:
+        return client.get_elastigroup(elastigroup_data.group_id)
+    except Exception:
+        current_span.set_tag('error', True)
+        current_span.log_kv({'exception': traceback.format_exc()})
+        logger.exception('Failed to get elastigroup {} {} in account {}'.format(elastigroup_data.group_name,
+                                                                                elastigroup_data.group_id,
+                                                                                elastigroup_data.account_id))
+
+
+@trace(pass_span=True)
+def get_elastigroup_instances(elastigroup_data, **kwargs):
+    """
+    Returns a list containing the active instances of an Elastigroup.
+    """
+    current_span = extract_span_from_kwargs(**kwargs)
+    current_span.set_tag("cloud_account_id", elastigroup_data.account_id)
+    current_span.set_tag("elastigroup_id", elastigroup_data.group_id)
+    current_span.set_tag("elastigroup_name", elastigroup_data.group_name)
+    current_span.set_tag("span.kind", "client")
+
+    client = SpotinstClient(auth_token=elastigroup_data.access_token, account_id=elastigroup_data.account_id,
+                            print_output=False)
+    try:
+        return client.get_elastigroup_active_instances(elastigroup_data.group_id)
```

Comment:
Is it possible we get throttled here by SpotInst? I would also suggest returning `[]`, as in case of exception, the caller assumes an iterator.